prime-router/src/test/kotlin/azure/FHIRFlowFunctionsTests.kt

```diff
@@ -112,4 +164,55 @@ badffffff9bffffffcb46fffffff5ffffff886fffffff84ffffff9efffffffaffffffd23bfffffff
             assertThat(fhirEngine.compare(case.input, case.output).passed).isEqualTo(case.matches)
         }
     }
+
+    @Test
+    fun `convert hl7 to fhir`() {
+        data class TestCase(
+            val name: String,
+            val input: String,
+            val entries: Int,
+            val status: HttpStatus,
+        )
+
+        val testCases = listOf(
+            TestCase(
+                "happy path",
+                testHL7NonBulk,
+                1,
+                HttpStatus.OK,
+            ),
+            TestCase(
+                "happy path - bulk",
+                testHL7Bulk,
+                3,
+                HttpStatus.OK,
+            ),
+            TestCase(
+                "bad hl7",
+                testBadHL7,
+                0,
+                HttpStatus.BAD_REQUEST
+            )
+        )
+
+        val fhirEngine = FHIRFlowFunctions()
+        testCases.forEach { case ->
+            val request = MockHttpRequestMessage(case.input)
+            val response = fhirEngine.convert(request)
+            assertThat(response.status).isEqualTo(case.status)
+            if (case.status == HttpStatus.OK) {
+                val body = response.body
+                check(body is String)
+                val entries = body.lines().filter { it.isNotEmpty() }
+                assertThat(entries.size).isEqualTo(case.entries)
+                entries.forEach { line ->
+                    try {
+                        FHIR.decode(line)
```

Comment:
So, no check on what was decoded, only that there was no error?  Will checks be added later?

Reply:
Correct: #4469